Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,565 @@
+            var activity = Activity.Current;
+
+            if (activity != null)
+            {
+                telemetry.Context.Operation.Id = activity.RootId;
+                telemetry.Context.Operation.ParentId = activity.ParentId;
+
+                foreach (var item in activity.Baggage)
+                {
+                    if (!telemetry.Context.Properties.ContainsKey(item.Key))
+                    {
+                        telemetry.Context.Properties[item.Key] = item.Value;
+                    }
+                }
+            }
+            else
+            {
+                telemetry.Context.Operation.Id = operationId.ToString("N");
+            }
+        }
+
+        private static void CorrelateErrorTelemetry(ExceptionTelemetry telemetry, Guid operationId)
+        {
+            telemetry.Context.Operation.ParentId = operationId.ToString("N");
+        }
+
+        private void OnAfterCommandEvent(
+            Guid operationId,
+            string operation,
+            Guid connectionId,
+            SqlCommand command,
+            IDictionary statistics,
+            long endTimestamp)
+        {
+            // Call DeriveDuration first to ensure we remove any start timestamp from operationStartTimestamps
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var dependencyName = string.Empty;
+            var target = string.Empty;
+
+            if (command.Connection != null)
+            {
+                target = string.Join(" | ", command.Connection.DataSource, command.Connection.Database);
+
+                var commandName = command.CommandType == CommandType.StoredProcedure
+                    ? command.CommandText
+                    : string.Empty;
+
+                dependencyName = string.IsNullOrEmpty(commandName)
+                    ? string.Join(" | ", command.Connection.DataSource, command.Connection.Database)
+                    : string.Join(" | ", command.Connection.DataSource, command.Connection.Database, commandName);
+            }
+
+            var telemetry = new DependencyTelemetry()
+            {
+                Id = operationId.ToString("N"),
+                Name = dependencyName,
+                Type = RemoteDependencyConstants.SQL,
+                Target = target,
+                Data = command.CommandText,
+                Duration = duration,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                Success = true
+            };
+
+            InitializeTelemetry(telemetry, operationId);
+            
+            this.client.Track(telemetry);
+        }
+
+        private void OnCommandErrorEvent(
+            Guid operationId,
+            string operation,
+            Guid connectionId,
+            Exception exception,
+            long endTimestamp)
+        {
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var telemetry = new ExceptionTelemetry(exception)
```

Comment:
Current dependency collectors only send DependencyTelemetry (success=false), and don't send any exception telemetry. This is major change and needs discussion with other stakeholders. We should follow the existing standards (i.e report DependencyTelemetry), and maybe add the exception code into the dependency telemetry itself, rather than sending a new ExceptionTelemetry. @SergeyKanzhelev